tests/test_structures.py

```diff
@@ -0,0 +1,75 @@
+# coding: utf-8
+import pytest
+
+from requests.structures import CaseInsensitiveDict, LookupDict
+
+
+class TestCaseInsensitiveDict:
+
+    @pytest.fixture
+    def dict_instance(self):
+        instance = CaseInsensitiveDict()
+        instance['Accept'] = 'application/json'
+        return instance
+
+    def test_list(self, dict_instance):
+        assert list(dict_instance) == ['Accept']
+
+    possible_keys = pytest.mark.parametrize('key', ('accept', 'ACCEPT', 'aCcEpT', 'Accept'))
+
+    @possible_keys
+    def test_getitem(self, dict_instance, key):
+        assert dict_instance[key] == 'application/json'
+
+    @possible_keys
+    def test_delitem(self, dict_instance, key):
+        del dict_instance[key]
+        assert key not in dict_instance
+
+    def test_lower_items(self, dict_instance):
+        assert list(dict_instance.lower_items()) == [('accept', 'application/json')]
+
+    def test_repr(self, dict_instance):
+        assert repr(dict_instance) == "{'Accept': 'application/json'}"
+
+    def test_copy(self, dict_instance):
+        copy = dict_instance.copy()
+        assert copy is not dict_instance
+        assert copy == dict_instance
+
+    @pytest.mark.parametrize(
+        'other, result', (
+            ({'AccePT': 'application/json'}, True),
+            ({}, False),
+            (None, False)
+        )
+    )
+    def test_equality(self, dict_instance, other, result):
+        assert (dict_instance == other) is result
+
+
+class TestLookupDict:
```

Comment:
Same feedback about this class. Use more descriptive function names and make these all independent functions please.

Alternatively, don't use fixtures passed into each method, but instead have a "set-up" fixture that runs for each method and defines the `instance` on `self`.

That would justify fairly bland test names grouped inside classes.